File: mlpack/nbc.cpp
#include <mlpack/core.hpp>
#include "naive_bayes_classifier.hpp"

using namespace mlpack;
using namespace mlpack::naive_bayes;

int main(int argc, char** argv)
{
    arma::mat data, labels, test;

    // Load the data.
    data::Load("../data/fisheriris_data.csv", data, true);
    test = data;
    data::Load("../data/fisheriris_label.csv", labels, true);
    data.insert_rows(data.n_rows, labels);
    
    // Naive Bayes classifier.
    NaiveBayesClassifier<> nbc(data, 4);
    
    // Predict labels.
    arma::Col<size_t> results;
    nbc.Classify(test, results);

    Log::cout << results << std::endl;    
    
    return 0;
}

File: shogun/nbc.py
import numpy as np
import os
from shogun.Features import RealFeatures, MulticlassLabels
from shogun.Classifier import GaussianNaiveBayes

# Load the data.
f = open(os.path.dirname(__file__) + '../data/fisheriris.data')
data = np.fromfile(f, dtype=np.float64, sep=' ')
data = data.reshape(-1, 4)
f.close()

f = open(os.path.dirname(__file__) + '../data/fisheriris_label.csv')
label = np.fromfile(f, dtype=np.float64, sep=' ')
f.close()

# Naive Bayes classifier.
feat = RealFeatures(data.T)
test = RealFeatures(data.T)
labels = MulticlassLabels(label)
nbc = GaussianNaiveBayes(feat, labels)
nbc.train()

# Predict labels.
results = nbc.apply(test).get_labels()

print results
